twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -247,26 +276,36 @@ def _compile_scala(
         unused_dependency_checker_ignored_targets = [],
     )
 
-    return JavaInfo(
-        source_jar = scrooge_jar,
-        deps = deps_java_info + implicit_deps,
-        runtime_deps = deps_java_info + implicit_deps,
-        exports = deps_java_info + implicit_deps,
-        output_jar = output,
-        compile_jar = output,
+    return _create_java_info_provider(scrooge_jar, all_deps, output)
+
+def _compile_generated_java(
+        ctx,
+        label,
+        output,
+        scrooge_jar,
+        deps_java_info,
+        implicit_deps):
+    all_deps = _concat_lists(deps_java_info, implicit_deps)
+    merged_deps = java_common.merge(all_deps)
+
+    compile_java(
+        ctx,
+        source_jars = [scrooge_jar],
+        source_files = [],
+        output = output,
+        extra_javac_opts = [],
+        providers_of_dependencies = [merged_deps],
     )
 
+    return _create_java_info_provider(scrooge_jar, all_deps, output)
+
 def _concat_lists(list1, list2):
     all_providers = []
     all_providers.extend(list1)
     all_providers.extend(list2)
     return all_providers
 
-####
-# This is applied to the DAG of thrift_librarys reachable from a deps
```

Comment:
why did you remove this comment?

Reply:
I meant to write a better one, but I didn't. My bad, will unremove.

Reply:
I merged that comment into a broader explanation of the aspect: https://github.com/bazelbuild/rules_scala/pull/1055/commits/ad320eb37c7ecf1e5245af070df601403bf96276#diff-4fe70c7ad76a7189b88d4792f15a77ffR363-R368